### ISSUE TITLE
[Feature] Packaged Electron app

### ISSUE THREAD

**Issue by DaniellMesquita** [Zarel/Pokemon-Showdown-Client#1157]

**Comment by Zarel** [comment]
We have a packaged NWjs app already. It will be ported to Electron in due time, but unless there's a specific Electron app feature you're asking for, this doesn't belong as an issue.

**Comment by DaniellMesquita** [comment]
The app is just a frame to the web version, and not good for slow connections

**Comment by Zarel** [comment]
...what do you think an Electron app is?

**Comment by DaniellMesquita** [comment]
> ...what do you think an Electron app is?

Most of Electron apps doesn't points to a external web site, but downloaded web app.
You earn money with Adsense, and you simply not want the risk of someone use this source and create a better web version

**Comment by Zarel** [comment]
Most Electron apps seem to be nearly identical to a downloaded web app. Discord being a good example.

(1) In any case, I plan to add service worker support to the Preact rewrite of the client, which will let the app download itself and work offline, while still auto-updating.

(2) And also in any case, that has nothing to do with Electron. Electron and NW.js and even service workers are the same in that they can support rendering from a local cached file or a downloaded remote file.

(3) And also, even without service workers, browsers can and do support caching, and PS has always taken advantage of that:

PS on first load:

![image](https://user-images.githubusercontent.com/551184/47821045-5f7b8f00-dd2d-11e8-9318-97951bbfd4fe.png)

PS on refresh:

![image](https://user-images.githubusercontent.com/551184/47821072-73bf8c00-dd2d-11e8-9fcb-916bd5ea5799.png)

A downloaded auto-updating app isn't going to be any less network-intensive. Offline apps are mainly useful because they allow working offline, not because they cache - browsers already cache without help.

(4) And also, this client is literally open-source. If someone wants to fork it and create their own version, no one is stopping them. Exactly what do you think I'm doing to _prevent_ someone from forking? I'd give the client a proprietary license if I actually cared about that.

In conclusion, you're extremely wrong about everything.
